diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/BracesInConditionalsAndLoopsRule.kt

```diff
@@ -0,0 +1,157 @@
+                    }
+                } else {
+                    val nodeAfterCondition = node.findChildByType(CONDITION)!!.nextSibling { it.elementType == ElementType.RPAR }!!.treeNext
+                    node.insertEmptyBlockBetweenChildren(nodeAfterCondition, nodeAfterCondition, indent)
+                }
+            }
+        }
+
+        if (hasElseBranch && elseNode?.elementType != IF && elseNode?.elementType != BLOCK) {
+            NO_BRACES_IN_CONDITIONALS_AND_LOOPS.warnAndFix(configRules, emitWarn, isFixMode, "ELSE",
+                    (elseNode?.treeParent?.prevSibling { it.elementType == ELSE_KEYWORD } ?: node).startOffset) {
+                if (elseNode != null) {
+                    ifPsi.`else`!!.replaceWithBlock(indent)
+                } else {
+                    // fixme: in which case else can have empty body?
+                    node.insertEmptyBlockBetweenChildren(node.findChildByType(ELSE_KEYWORD)!!, null, indent)
+                }
+            }
+        }
+    }
+
+    private fun isSingleLineIfElse(node: ASTNode, elseNode: ASTNode?): Boolean {
+        val hasSingleElse = elseNode != null && elseNode.elementType != IF
+        return node.treeParent.elementType != ELSE && hasSingleElse && node.text.lines().size == 1
+    }
+
+    private fun checkLoop(node: ASTNode) {
+        val loopBody = (node.psi as KtLoopExpression).body
+        val loopBodyNode = loopBody?.node
+        if (loopBodyNode == null || loopBodyNode.elementType != BLOCK) {
+            NO_BRACES_IN_CONDITIONALS_AND_LOOPS.warnAndFix(configRules, emitWarn, isFixMode, node.elementType.toString(), node.startOffset) {
+                // fixme proper way to calculate indent? or get step size (instead of hardcoded 4)
+                val indent = node.prevSibling { it.elementType == WHITE_SPACE }!!.text.lines().last().count { it == ' ' }
+                if (loopBody != null) {
+                    loopBody.replaceWithBlock(indent)
+                } else {
+                    // this corresponds to do-while with empty body
+                    node.insertEmptyBlockBetweenChildren(
+                            node.findChildByType(DO_KEYWORD)!!.treeNext,
+                            node.findChildByType(WHILE_KEYWORD)!!.treePrev,
+                            indent
+                    )
+                }
+            }
+        }
+    }
+
+    private fun checkWhenBranches(node: ASTNode) {
+        (node.psi as KtWhenExpression).entries.asSequence()
+                .filter { it.expression != null && it.expression!!.node.elementType == BLOCK }
+                .map { it.expression as KtBlockExpression }
+                .filter { it.statements.size == 1 }
+                .forEach {
+                    NO_BRACES_IN_CONDITIONALS_AND_LOOPS.warnAndFix(configRules, emitWarn, isFixMode, "WHEN", it.node.startOffset) {
+                        it.astReplace(it.firstStatement!!.node.psi)
```

Comment:
don't get it, where do you add braces?

Reply:
I don't, for `when` statements we decided to only remove braces for single-line branches, didn't we?

Reply:
ah, yeah, really, forgot it